lib/plain/entities/workflow-definition.ts

```diff
@@ -0,0 +1,218 @@
+import { RawAxiosRequestHeaders } from 'axios'
+import {
+  GetWorkflowDefinitionParams,
+  GetSpaceEnvironmentParams,
+  CollectionProp,
+} from '../../common-types'
+import {
+  WorkflowDefinitionProps,
+  WorkflowDefinitionQueryOptions,
+  CreateWorkflowDefinitionParams,
+  CreateWorkflowDefinitionProps,
+  UpdateWorkflowDefinitionParams,
+  UpdateWorkflowDefinitionProps,
+  DeleteWorkflowDefinitionParams,
+} from '../../export-types'
+import { OptionalDefaults } from '../wrappers/wrap'
+
+export type WorkflowDefinitionPlainClientAPI = {
+  /**
+   * Fetch a Workflow Definition
+   * @param params entity IDs to identify the Workflow Definition
+   * @returns the Workflow Definition
+   * @throws if the request fails, or the Space/Environment is not found
+   * @example
+   * ```javascript
+   * const workflowDefinition = await client.workflowDefinition.get({
+   *   spaceId: '<space_id>',
+   *   environmentId: '<environment_id>',
+   *   workflowDefinitionId: '<workflow_definition_id>',
+   * });
+   * ```
+   */
+  get(
+    params: OptionalDefaults<GetWorkflowDefinitionParams>,
+    headers?: RawAxiosRequestHeaders
+  ): Promise<WorkflowDefinitionProps>
+  /**
+   * Query Workflow Definitions with certain filters
+   * @param params entity IDs to identify the Space/Environment, optional query parameters to filter returned Workflow Definitions
+   * @returns an object containing the list of Workflow Definitions
+   * @throws if the request fails, or the Space/Environment is not found
+   * @example
+   * ```javascript
+   * const workflowDefinitions = await client.workflowDefinition.getMany({
+   *   spaceId: '<space_id>',
+   *   environmentId: '<environment_id>',
+   *   query: {
+   *     limit: 10,
+   *   }
+   * });
+   * ```
+   * */
+  getMany(
+    params: OptionalDefaults<
+      GetSpaceEnvironmentParams & { query?: WorkflowDefinitionQueryOptions }
+    >,
+    headers?: RawAxiosRequestHeaders
+  ): Promise<CollectionProp<WorkflowDefinitionProps>>
+  /**
+   * Create a new Workflow Definition
+   * @param params entity IDs to identify the Space/Environment to create the Workflow Definition in
+   * @param rawData the new Workflow Definition
+   * @throws if the request fails, or the Space/Environment is not found
+   * @example
+   * ```javascript
```

Comment:
just wondering what this snippet will look like when you hover over this type in an editor, might be kinda hard to read. what does the legacy client example look like for this operation?